src/services/groups.ts

```diff
@@ -30,25 +42,92 @@ interface GrantsResponse {
   status: string;
 }
 
+interface GroupCollaboratorsResponse {
+  id: number;
+  collaboratorType: { name: string };
+  user: { id: number, name: string };
+}
+
 interface GroupResponse {
   id: number;
   name: string;
   grants: GrantsResponse[];
-  userId: number;
+  groupCollaborators: GroupCollaboratorsResponse[];
   isPublic: boolean;
 }
 
-export async function groupsByRegion(region: number): Promise<GroupResponse[]> {
+/**
+ * Checks if a group name is available.
+ *
+ * @param name - The name of the group to check.
+ * @param groupId - Optional. The ID of the group to exclude from the check.
+ * @returns A Promise that resolves to a boolean value indicating if the group name is available
+ * (true) or not (false).
+ */
+export async function checkGroupNameAvailable(name: string, groupId?: number): Promise<boolean> {
+  console.log({ name, groupId });
+  // Find an existing group with the given name and optional groupId
+  const existingGroup = await Group.findOne({
+    attributes: ['id'],
+    where: {
+      name: {
+        [Op.iLike]: name.trim(), // Case-insensitive comparison of the name
+      },
+      // Exclude the group with the given groupId if provided
+      ...(groupId && { groupId: { [Op.not]: groupId } }),
+    },
+  });
+  console.log(existingGroup);
+  return !!existingGroup; // Return true if an existing group is found, false otherwise
+}
+
+/**
+ * Retrieves groups by region.
+ *
+ * @param region - The region ID to filter the groups by.
+ * @returns A promise that resolves to an array of GroupResponse objects.
+ * @throws This function does not throw any exceptions.
+ */
+export async function groupsByRegion(userId: number, region: number): Promise<GroupResponse[]> {
+  // TODO: needs to take current userID as an arg so the permissions can be checked in the query
   return Group.findAll({
     attributes: [
       'id',
       'name',
-      'userId',
```

Comment:
@GarrettEHill I'm starting to update this code. What is the idea here for GroupGrants are we also creating the userId column in that table? What about group types?

Reply:
So after looking at this again I guess the idea would be we pull from group collaborators then use the 'groupId' to get any linked grants.